### PR TITLE
fix: [CDS-71859]: fixed the InputSet reconcile API call

### PR DESCRIPTION
**Pull request by sahithikolichala**
### Summary

Alligned with the changed BE API contract and clean up:  removed conditional query param for allow_different_repo_setting and now always sending  : `pipelineBranch` ,
`branch` : **inputSetBranch**  if availble.

BE support PR : https://github.com/harness/harness-core/pull/48990

**Use case Tested for**

- Inline : working
- GItX : Input set List page ( Input set branch not available)
      Not working : same repo same branch ( tested for default branch)
      Working with expected error  - same repo other branch
      Working -  other repo default branch
      Working with expected error -  other repo feature branch

- GItX : Input set  Details page ( Input set branch available)
    Working -   same repo same branch
    Working with expected error  - same repo other branch
    Working -  other repo default branch
    Working with expected error -  other repo feature branch  ( Even UI sends IS branch as feature, BE try to reconcile with default branch)

- Old git sync
    Same repo : working
    Other repo :  UI is sending expected, BE need to fix.

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by sahithikolichala** [comment]
retrigger titlecheck

**Comment by sahithikolichala** [comment]
retrigger cypress-pipeline

**Review by Sagar-Rajput7 — APPROVED** [review]

File: src/modules/70-pipeline/components/InputSetErrorHandling/OutOfSyncErrorStrip/OutOfSyncErrorStrip.tsx [hunk 1]
```diff
@@ -186,7 +186,9 @@ export function OutOfSyncErrorStrip(props: OutOfSyncErrorStripProps): React.Reac
 
       connectorRef: connectorRef ?? get(inputSet, 'connectorRef'),
       storeType: storeType ?? get(inputSet, 'storeType'),
-      ...gitParams
+      ...gitParams,
+      pipelineBranch: reconcileBranch,
```
Comment (Sagar-Rajput7) [inline src/modules/70-pipeline/components/InputSetErrorHandling/OutOfSyncErrorStrip/OutOfSyncErrorStrip.tsx:190]:
```
      ...(isGitSyncEnabled
        ? {
            pipelineRepoID: repoIdentifier,
            pipelineBranch: branch
          }
        : {}),
```


similar lines are defined above `174` but behind - isGitSyncEnabled, please check if we want to show these here always regardless of `isGitSyncEnabled`

Comment (sahithikolichala) [reply src/modules/70-pipeline/components/InputSetErrorHandling/OutOfSyncErrorStrip/OutOfSyncErrorStrip.tsx:190]: Thanks for catching this @Sagar-Rajput7  
I will check it out

Comment (sahithikolichala) [reply src/modules/70-pipeline/components/InputSetErrorHandling/OutOfSyncErrorStrip/OutOfSyncErrorStrip.tsx:190]: @Sagar-Rajput7  ideally we have to pass , when gitsync is enabled 
for this pipeline 
https://app.harness.io/ng/account/SNM_3IzhRa6SFPz6DIV7aA/home/orgs/default/projects/testrohitgitops/pipelines/testgitsyncprod/input-sets?connectorRef=rohitgithubtoken&repoName=rohittest&branch=git-sync-test&storeType=REMOTE

isGitSyncEnabled is false always which is not right

File: src/modules/70-pipeline/components/InputSetErrorHandling/OutOfSyncErrorStrip/OutOfSyncErrorStrip.tsx [hunk 2]
```diff
@@ -171,12 +171,7 @@ export function OutOfSyncErrorStrip(props: OutOfSyncErrorStripProps): React.Reac
       projectIdentifier,
       orgIdentifier,
       pipelineIdentifier: pipelineIdentifier ?? get(inputSet, 'pipelineIdentifier'),
-      ...(isGitSyncEnabled
-        ? {
-            pipelineRepoID: repoIdentifier,
```
Comment (sunnykesh-harness) [inline src/modules/70-pipeline/components/InputSetErrorHandling/OutOfSyncErrorStrip/OutOfSyncErrorStrip.tsx:176]: Has this got removed in API contract for old gitSync?

**Review by sunnykesh-harness — COMMENTED** [review]
Please list down the use case tested with this change in PR description.

File: src/modules/70-pipeline/components/InputSetErrorHandling/OutOfSyncErrorStrip/OutOfSyncErrorStrip.tsx [hunk 3]
```diff
@@ -171,22 +157,19 @@ export function OutOfSyncErrorStrip(props: OutOfSyncErrorStripProps): React.Reac
...
-            pipelineBranch: branch
-          }
-        : {}),
       repoIdentifier: isGitSyncEnabled
         ? overlayInputSetRepoIdentifier ?? inputSetRepoIdentifier
         : repoName ?? get(inputSet, 'gitDetails.repoName'),
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/components/InputSetErrorHandling/OutOfSyncErrorStrip/OutOfSyncErrorStrip.tsx:162]: Do we no more need to consider `ALLOW_DIFFERENT_REPO_FOR_INPUT_SETS` project setting?

Comment (sunnykesh-harness) [reply src/modules/70-pipeline/components/InputSetErrorHandling/OutOfSyncErrorStrip/OutOfSyncErrorStrip.tsx:162]: no, we are making simple and consistent contract.

**Review by Eric761 — APPROVED** [review]

**Review by Sagar-Rajput7 — APPROVED** [review]

**Pull request merged**